### PR TITLE
PostgreSQL Support

### LINKED ISSUE #2025: Add PostgreSQL support
Please add PostgreSQL support in the panel for consistency with other projects.

PostgreSQL could be natively implemented into the panel, or using the [mysql_fdw](https://github.com/EnterpriseDB/mysql_fdw) wrapper to convert MySQL queries into PostgreSQL queries.

### PR DESCRIPTION
**Pull request by lancepioch**
@matthewpi There's 2 migrations that still need one json function each swapped out to the postgresql equivalent. They are marked with TODO

### CODE REVIEW

**Review by matthewpi — APPROVED** [review]

File: database/migrations/2021_01_17_152623_add_generic_server_status_column.php [hunk 1]
```diff
@@ -9,20 +9,16 @@ class AddGenericServerStatusColumn extends Migration
...
             $table->string('status')->nullable()->after('description');
         });
 
-        DB::transaction(function () {
-            DB::update('UPDATE servers SET `status` = \'suspended\' WHERE `suspended` = 1');
-            DB::update('UPDATE servers SET `status` = \'installing\' WHERE `installed` = 0');
-            DB::update('UPDATE servers SET `status` = \'install_failed\' WHERE `installed` = 2');
-        });
+        DB::table('servers')->where('suspended', 1)->update(['status' => 'suspended']);
+        DB::table('servers')->where('installed', 1)->update(['status' => 'installing']);
```
Comment (tryvin) [inline database/migrations/2021_01_17_152623_add_generic_server_status_column.php:20]: @lancepioch Small typo here, `installed = 0` means `status = installing`

File: database/migrations/2017_08_05_144104_AllowNegativeValuesForOverallocation.php [hunk 2]
```diff
@@ -20,10 +20,10 @@ public function up()
     /**
      * Reverse the migrations.
      */
-    public function down()
+    public function down(): void
     {
         Schema::table('nodes', function (Blueprint $table) {
-            DB::statement('ALTER TABLE nodes MODIFY disk_overallocate MEDIUMINT UNSIGNED NULL, 
+            DB::statement('ALTER TABLE nodes MODIFY disk_overallocate MEDIUMINT UNSIGNED NULL,
```
Comment (tryvin) [inline database/migrations/2017_08_05_144104_AllowNegativeValuesForOverallocation.php:26]: @lancepioch Shouldn't this be the usual `$table->change`?

File: database/migrations/2016_10_23_201624_add_foreign_allocations.php [hunk 3]
```diff
@@ -25,14 +23,17 @@ public function up()
...
+            $table->dropForeign(['node']);
+            $table->dropIndex(['node']);
 
-            $table->dropIndex('allocations_assigned_to_foreign');
-            $table->dropIndex('allocations_node_foreign');
+            $table->mediumInteger('assigned_to', false, true)->nullable()->change();
+            $table->mediumInteger('node', false, true)->nullable(false)->change();
         });
 
         DB::statement('ALTER TABLE allocations
```
Comment (tryvin) [inline database/migrations/2016_10_23_201624_add_foreign_allocations.php:39]: @lancepioch Shouldn't this be the `$table->change`?

**Comment by tryvin** [comment]
@lancepioch Sorry to dig this PR from the grave, but I was looking to add Cockroach DB (PostgreSQL-based database) support to pterodactyl and found that you started adding Postgres support a couple of months ago, so as I was reading the changes I noticed these small things.

If you can't do them, do you want me to make these fixes?

**Pull request merged**